crates/ruff_linter/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -71,6 +75,19 @@ use crate::{Edit, Fix, FixAvailability, Violation};
 /// print("{}".format(val))  # "(1,)"
 /// ```
 ///
+/// It is also possible for the right-hand side to have values that implement
+/// `__format__` to not behave the same as `__str__` when `None` is passed:
```

Comment:
Not sure exactly the best phrasing here, maybe something like this?

```suggestion
/// Additionally, while `__format__` typically calls `__str__`, this is not always
/// the case. If `__format__` has been overridden, replacing `%s` with the
/// empty `format` placeholder will change the program's behavior:
```

I also think we should mention explicitly when the fix _is_ safe. To me that feels most natural right at the beginning of the section ("The fix is only marked as safe when the value being formatted is a literal"). We should also try to work `In preview, ...` in there somewhere.